README.md

```diff
@@ -213,6 +213,22 @@ export default Eng;
 Currently, only services and route paths (see below) can be shared across the
 parent/engine boundary.
 
+If you want to share components, helpers or utils between a parent and
+an engine, then create an addon and add it as a dependency of both
+your application and engine.
+
+In-repo addons can be consumed from an engine. To do so, add it to the
+engine's `package.json`:
+
+```
+// lib/my-in-repo-engine/package.json
+  "ember-addon": {
+    "paths": [
+      "../shared-components"
```

Comment:
@nathanhammond updated. Added as a section with both scenarios.

I noticed there is some duplication between the README and articles under guides. Maybe we can publish stuff under guides as a github page and leave the README for dev notes only? cc @knownasilya